theano/gpuarray/dnn.py

```diff
@@ -1384,45 +1384,47 @@ def make_node(self, dy, sm):
 
 @local_optimizer([AbstractConv2d, AbstractConv2d_gradWeights,
                   AbstractConv2d_gradInputs])
-def local_abstractconv_cudnn(node):
-    if (not isinstance(node.op, (AbstractConv2d,
+@register_opt2([AbstractConv2d, AbstractConv2d_gradWeights,
+                  AbstractConv2d_gradInputs], 'fast_compile')
+def local_abstractconv_cudnn(op, context_name, inputs):
```

Comment:
This interface won't work with the old back-end.

You can do this 2 functions:

 @local_optimizer([AbstractConv2d, AbstractConv2d_gradWeights,
                   AbstractConv2d_gradInputs])
                   AbstractConv2d_gradInputs])
def local_abstractconv_cudnn(node):
    ctx = infer_context(*node.inputs)
    return local_abstractconv_dnn_graph(node.op, ctx, node.inputs)

@register_opt2([AbstractConv2d, AbstractConv2d_gradWeights,
                  AbstractConv2d_gradInputs], 'fast_compile')
def local_abstractconv_cudnn_graph(op, context_name, inputs):
    The current version of local_absstraconv_cudnn